### PR TITLE
Microplan search

### PR DESCRIPTION
**Pull request by abishekTa-egov**
Able to search by name of microplan

### CODE REVIEW

File: health/micro-ui/web/micro-ui-internals/packages/modules/microplan/src/configs/UICustomizations.js [hunk 1]
```diff
@@ -10,108 +12,193 @@ const businessServiceMap = {};
...
-          });
-          return link;
-        },
-        additionalValidations: (type, data, keys) => {
-          if (type === "date") {
-            return data[keys.start] && data[keys.end] ? () => new Date(data[keys.start]).getTime() <= new Date(data[keys.end]).getTime() : true;
-          }
-        },
-      },
+  SearchPucarConfig: {
```
Comment (nipunarora-eGov) [inline health/micro-ui/web/micro-ui-internals/packages/modules/microplan/src/configs/UICustomizations.js:37]: @abishekTa-egov remove this SearchPucarConfig object

File: health/micro-ui/web/micro-ui-internals/packages/modules/microplan/src/configs/UICustomizations.js [hunk 2]
```diff
@@ -10,108 +12,193 @@ const businessServiceMap = {};
...
+      data.body.PlanConfigurationSearchCriteria.limit = data?.state?.tableForm?.limit;
+      // data.body.PlanConfigurationSearchCriteria.limit = 10
+      data.body.PlanConfigurationSearchCriteria.offset = data?.state?.tableForm?.offset;
+      data.body.PlanConfigurationSearchCriteria.name = name;
+      data.body.PlanConfigurationSearchCriteria.tenantId = Digit.ULBService.getCurrentTenantId();
+      data.body.PlanConfigurationSearchCriteria.userUuid = Digit.UserService.getUser().info.uuid;
+      // delete data.body.PlanConfigurationSearchCriteria.pagination
+      data.body.PlanConfigurationSearchCriteria.status = status?.status;
+      cleanObject(data.body.PlanConfigurationSearchCriteria);
+      // debugger;
```
Comment (nipunarora-eGov) [inline health/micro-ui/web/micro-ui-internals/packages/modules/microplan/src/configs/UICustomizations.js:155]: remove debuggers and console.logs

File: health/micro-ui/web/micro-ui-internals/packages/modules/microplan/src/pages/employee/MicroplanSearch.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+const MicroplanSearch = () => {
+  const { t } = useTranslation();
+  const [defaultValues, setDefaultValues] = useState(defaultSearchValues); // State to hold default values for search fields
+  const [config, setConfig] = useState(TabSearchconfig?.TabSearchconfig?.[0]); // initially setting first index config as default from jsonarray
+  const [tabData, setTabData] = useState(
+    TabSearchconfig?.TabSearchconfig?.map((configItem, index) => ({ key: index, label: configItem.label, active: index === 0 ? true : false }))
+  ); // setting number of tab component and making first index enable as default
+  useEffect(() => {
+    // Set default values when component mounts
+    console.log("hihihiijjiji");
```
Comment (nipunarora-eGov) [inline health/micro-ui/web/micro-ui-internals/packages/modules/microplan/src/pages/employee/MicroplanSearch.js:22]: logs

File: health/micro-ui/web/micro-ui-internals/packages/modules/microplan/src/pages/employee/MicroplanSearch.js [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    setDefaultValues(defaultSearchValues);
+  }, []);
+
+  const onTabChange = (n) => {
+    debugger
```
Comment (nipunarora-eGov) [inline health/micro-ui/web/micro-ui-internals/packages/modules/microplan/src/pages/employee/MicroplanSearch.js:27]: debugger

File: health/micro-ui/web/micro-ui-internals/packages/modules/microplan/src/pages/employee/MicroplanSearch.js [hunk 5]
```diff
@@ -0,0 +1,52 @@
+import { Header, InboxSearchComposer } from "@egovernments/digit-ui-react-components";
+import React, { useState, useEffect } from "react";
+import { useTranslation } from "react-i18next";
+import { TabSearchconfig } from '../../configs/MicroplanSearchConfig'
+
+
+
+const defaultSearchValues = {
+  individualName: "",
```
Comment (nipunarora-eGov) [inline health/micro-ui/web/micro-ui-internals/packages/modules/microplan/src/pages/employee/MicroplanSearch.js:9]: remove unneccesary code

File: health/micro-ui/web/micro-ui-internals/packages/modules/microplan/src/configs/MicroplanSearchConfig.js [hunk 6]
```diff
@@ -0,0 +1,127 @@
+
+
+// default values of search input component
+const defaultSearchValues = {
```
Comment (nipunarora-eGov) [inline health/micro-ui/web/micro-ui-internals/packages/modules/microplan/src/configs/MicroplanSearchConfig.js:4]: why this is required?

File: health/micro-ui/web/micro-ui-internals/packages/modules/microplan/src/configs/MicroplanSearchConfig.js [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+                  //     name: "individualId",
+                  //   },
+                  // },
+                ],
+              },
+    
+              show: true,
+            },
+            searchResult: {
+              tenantId: Digit.ULBService.getCurrentTenantId(),
```
Comment (nipunarora-eGov) [inline health/micro-ui/web/micro-ui-internals/packages/modules/microplan/src/configs/MicroplanSearchConfig.js:73]: this should be simply json without any functions

File: health/micro-ui/web/micro-ui-internals/packages/modules/microplan/src/services/searchSavedPlans.js [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+        method: "POST",
+        userService: false,
+        body: {
+          CampaignDetails,
+        },
+      });
+  
+      const finalResult = {
+        PlanConfiguration: mergeArrays(responsePlan?.PlanConfiguration, "executionPlanId", responseCampaign?.CampaignDetails, "id"),
+      };
```
Comment (nipunarora-eGov) [inline health/micro-ui/web/micro-ui-internals/packages/modules/microplan/src/services/searchSavedPlans.js:60]: debugger

**Review by nipunarora-eGov — CHANGES_REQUESTED** [review]
Pls check comments.

**Comment by abishekTa-egov** [comment]
ok will make the changes

**Comment by abishekTa-egov** [comment]
Changes everything except the default values, since may want to enter in future, if needed i will remove

**Comment by nipunarora-eGov** [comment]
> Changes everything except the default values, since may want to enter in future, if needed i will remove

Here just mention the object as default values inside config don't create reference, keep your config file as plain JSON object

**Review by nipunarora-eGov — APPROVED** [review]

**Pull request merged**